viewing_party/party.py

```diff
@@ -1,23 +1,174 @@
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    for key, value in new_movie.items():
+        print(new_movie)
+        if value == None:
+            return None
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"] = []
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    # for movies["title"] in user_data["watched"] == title:
+    for movies in user_data["watchlist"]:
+        if movies["title"] == title:
+            print(title)
+            user_data["watched"].append(movies)
+            user_data["watchlist"].remove(movies)
+    return user_data
+        
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0
+    length = len(user_data["watched"])
+    if length==0:
+        return 0
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+    avg = sum/ length
+    return avg            
+
+def get_most_watched_genre(user_data):
+    genre_count = {}
+    # if user_data["watched"]==[]:
+    # length = len(user_data["watched"])
+    # if length==0:
+    #     return None
+    if not user_data["watched"]:
+        return None 
+    for movie in user_data["watched"]:
+        genre = movie["genre"] 
+        if genre not in genre_count:
+            genre_count[genre] = 1
+        else:
+            genre_count[genre] += 1
+    sorted_genre = sorted(genre_count.items(), key = lambda kv:kv[1])
+    return sorted_genre[-1][0]
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def friends_list(user_data):
+    friends_movie_list = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in friends_movie_list:
+                friends_movie_list.append(movie)
+    return friends_movie_list
+
+def users_list(user_data):
+    users_movie_list= []
+    for movie in user_data["watched"]:
+        if movie not in users_movie_list:
+            users_movie_list.append(movie)
+    return users_movie_list
+
+def get_unique_watched(user_data):
+    user_unique_list=[]
+    friends_movies = friends_list(user_data)
+    users_movies = users_list(user_data)
+   
+    for movie in users_movies:
+        if movie not in friends_movies:
+            user_unique_list.append(movie)
+    return user_unique_list
+    # print(unique_list == friends_movies - users_movies)
+    # return unique_list == friends_movies - users_movies 
+    # user_unique_list == list(set(friends_movies)-set(users_movies))     
+    # return user_unique_list

```

Comment:
I think I see what you're trying to do here by casting lists into sets and then getting the difference between them. 

When you cast something into a set, the elements need to be **immutable**, however right now what you have on like line 114, you're casting a list of dictionaries into a set. Dictionaries are mutable and cannot be in sets. 

If you want to get the difference between sets, you'll need friends_movies and users_movies to be lists of titles that are strings (not the entire movie dictionary). After you get the difference from the 2 sets that have titles in them, then you'll need a for loop to find all the dictionaries with title in the difference and build a new list with dictionaries. 

```python
friends_movies = get_friends_watch_list(user_data)  # this would need to be a list of strings
users_movies = get_user_watch_list(user_data) # this would need to be a list of strings

diff = set(users_movies) - set(friends_movies)
unique_list = list(diff)

unique_dict = []
    for movie in user_data["watched"]:
        if movie["title"] in unique_list:
            unique_dict.append(movie)

    return unique_dict
```